### PR TITLE
enhancement: Added 'errors=coerce' option in astype()

### LINKED ISSUE #48781: ENH: add errors='coerce' to DataFrame.astype
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I wish I could quickly convert a DataFrame with some invalid data to numeric type and coerce. I thought `pd.DataFrame.astype` could do that but it doesn't have the option to coerce invalid data to NaNs (or NaTs).

In my particular case I have a DataFrame of sensor readings with mostly NaNs (indicating no value received), many integers (those I care about), and some strings (indicating specific errors). I quickly tried to get a histogram to get an overview of that data but the `pd.DataFrame.hist` requires numeric data which is a few lines of code to get. This is exploratory code I write in my console, so it would be sweet if this could be done with a single method.

### Toy Example

```
import numpy as np
import pandas as pd

df = pd.DataFrame([
    [np.NaN, 0.1, 1.1, 1.6],
    ["error", 0.2, 1.2, 1.7],
    [0.3, "", 1.3, 1.8],
    [0.4, 1.4, "code255", 1.9],
])

df.astype(float, errors="coerce")
# ValueError: Expected value of kwarg 'errors' to be one of ['raise', 'ignore'].
# Supplied value is 'coerce'

import matplotlib.pyplot as plt
plt.hist(df.values.flatten(), bins=[0, 1, 2])
```

Expected result:

```
In [30]: df
Out[30]: 
     0    1    2    3
0  NaN  0.1  1.1  1.6
1  NaN  0.2  1.2  1.7
2  0.3  NaN  1.3  1.8
3  0.4  1.4  NaN  1.9
```
image

### Feature Description

Two options:

* Allow multidimensional input (e.g. DataFrames, Numpy Arrays) as the `arg` of `pd.to_numeric`. In case of mixed type columns (e.g. integers and floats), we'd have to decide and document if that would operate by column or cast the whole data structure to one dtype. I'd expect by column for DataFrames. Another issue that comes up is how to deal with multidimensional lists and tuples.

OR/AND

* Add the option `"coerce"` to the `errors` kwarg in `pd.DataFrame.astype` (the current options are `"raise"` and `"ignore"`. We'd have to decide how to deal with incompatibilities between `errors="coerce"` and `dtype`. E.g. what to do if someone tries to coerce to string. I would expect an error.

To me it looks like the potential for confusing the user is a lot lower with the second option because it has fewer edge cases.

### Alternative Solutions

```
for col in df.cols:
    df[col] = pd.to_numeric(df[col], errors="coerce")
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by subbusainath**
- [ ] closes #48781  
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [errors=coerce in astype()](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/v1.6.1.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
tests always

**Review by mroeschke — CHANGES_REQUESTED** [review]
Thanks for the pull request. The original issue hasn't had much discussion or agreement from the core team, so I would recommend waiting until there's more consensus

**Comment by mroeschke** [comment]
Going to close for now pending the discussion in the original issue. Happy to have your thoughts in that issue if you are in support of this feature.

**Comment by subbusainath** [comment]
Why didn't we moved this PR ? Since it does not have a huge enhancement in the astype method. I would like to add this enhancement feature into astype (). Which help us to solve the issue. I hope!

**Comment by mroeschke** [comment]
We would still be open to this enhancement given more support and discussion in #48781 first.

**Comment by mroeschke** [comment]
Actually to stimulate discussion, I'll reopen this PR, sorry.

File: doc/source/whatsnew/v1.6.1.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
+.. _whatsnew_161:
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.6.1.rst:1]: if accepted, this would go in the 1.6.0 file

File: pandas/core/dtypes/astype.py [hunk 2]
```diff
@@ -302,6 +303,8 @@ def astype_array_safe(
         #  trying to convert to float
         if errors == "ignore":
             new_values = values
+        elif errors == "coerce":
+            new_values = values
```
Comment (jbrockmendel) [inline pandas/core/dtypes/astype.py:307]: how is this different from `errors="ignore"`?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. Additionally this issue could use more discussion on the original issue before moving forward to closing to await resolution on the main issue

**Pull request closed without merging**